### PR TITLE
Fix some buggy/useless usages of stripslashes

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

There are still many usages of addslashes/stripslashes in codebase, and it may produce bug on edge cases (usage of namespaced itemtypes for instance). I fixed some of them.

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

File: ajax/impact.php [hunk 1]
```diff
@@ -120,7 +120,7 @@
         }
 
        // Decode data (should be json)
-        $data = Toolbox::jsonDecode($_UPOST['impacts'], true);
+        $data = Toolbox::jsonDecode($_POST['impacts'], true);
```
Comment (trasher) [inline ajax/impact.php:123]: Are you sure for this one? I had in mind JSON causes issues when used from `$_POST`

Comment (cedric-anne) [reply ajax/impact.php:123]: I have modified the `Toolbox::jsonDecode()` to ensure sanitization process is correctly reverted, and I added a test case for this.

Comment (trasher) [reply ajax/impact.php:123]: ok :)
